akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRecoveringEvents.scala

```diff
@@ -0,0 +1,222 @@
+/**
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.persistence.typed.internal
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors.MutableBehavior
+import akka.actor.typed.scaladsl.{ ActorContext, Behaviors, StashBuffer, TimerScheduler }
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.persistence.JournalProtocol._
+import akka.persistence._
+import akka.persistence.typed.internal.EventsourcedBehavior.WriterIdentity
+import akka.persistence.typed.scaladsl.PersistentBehaviors._
+import akka.util.Helpers._
+
+import scala.util.control.NonFatal
+
+/**
+ * INTERNAL API
+ *
+ * Third (of four) behavior of an PersistentBehavior.
+ *
+ * In this behavior we finally start replaying events, beginning from the last applied sequence number
+ * (i.e. the one up-until-which the snapshot recovery has brought us).
+ *
+ * Once recovery is completed, the actor becomes [[EventsourcedRunning]], stashed messages are flushed
+ * and control is given to the user's handlers to drive the actors behavior from there.
+ *
+ */
+@InternalApi
+private[akka] class EventsourcedRecoveringEvents[Command, Event, State](
+  val persistenceId:          String,
+  override val context:       ActorContext[Any],
+  override val timers:        TimerScheduler[Any],
+  override val internalStash: StashBuffer[Any],
+
+  val recovery:           Recovery,
+  private var sequenceNr: Long,
+  val writerIdentity:     WriterIdentity,
+
+  private var state: State,
+
+  val callbacks: EventsourcedCallbacks[Command, Event, State],
+  val pluginIds: EventsourcedPluginIds
+) extends MutableBehavior[Any]
+  with EventsourcedBehavior[Command, Event, State]
+  with EventsourcedStashManagement {
+
+  import Behaviors.same
+  import EventsourcedBehavior._
+  import akka.actor.typed.scaladsl.adapter._
+
+  protected val log = Logging(context.system.toUntyped, this)
+
+  // -------- initialize --------
+  startRecoveryTimer()
+
+  replayEvents(sequenceNr + 1L, recovery.toSequenceNr)
+  // ---- end of initialize ----
+
+  private def commandContext: ActorContext[Command] = context.asInstanceOf[ActorContext[Command]]
```

Comment:
We accept ActorContext[Any] which someone else has to force-typecast and then we typecast it back, can't it bee ActorContext[Command] throughout?